Classes/iOSDeviceManagerLogging.m

```diff
@@ -0,0 +1,230 @@
+}
+
+
+- (NSDateFormatter *)logFileDateFormatter {
+  NSMutableDictionary *dictionary = [[NSThread currentThread] threadDictionary];
+  NSString *dateFormat = IDMLogFileNameDateFormat;
+  NSDateFormatter *dateFormatter = dictionary[IDMLogFileNameDateFormatterKey];
+
+  if (dateFormatter == nil) {
+    dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_US_POSIX"]];
+    [dateFormatter setDateFormat:dateFormat];
+    [dateFormatter setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
+    dictionary[IDMLogFileNameDateFormatterKey] = dateFormatter;
+  }
+
+  return dateFormatter;
+}
+
+- (NSString *)createNewLogFile {
+  NSString *fileName = [self newLogFileName];
+  NSString *logsDirectory = [self logsDirectory];
+
+  NSString *path = [logsDirectory stringByAppendingPathComponent:fileName];
+  while ([[NSFileManager defaultManager] fileExistsAtPath:path]) {
+    path = [logsDirectory stringByAppendingPathComponent:[self newLogFileName]];
+  }
+
+  NSDictionary *attributes = nil;
+
+  [[NSFileManager defaultManager] createFileAtPath:path
+                                          contents:nil
+                                        attributes:attributes];
+
+
+  [self deleteOldLogFiles];
+
+  NSString *currentSymlink = [logsDirectory stringByAppendingPathComponent:@"current.log"];
+  [[NSFileManager defaultManager] removeItemAtPath:currentSymlink error:nil];
+
+  [[NSFileManager defaultManager] createSymbolicLinkAtPath:currentSymlink
+                                       withDestinationPath:path error:nil];
+
+  return path;
+}
+
+- (BOOL)isLogFile:(NSString *)fileName {
+
+  BOOL hasProperPrefix = [fileName hasPrefix:IDMLogFilePrefix];
+  BOOL hasProperSuffix = [fileName hasSuffix:@".log"];
+  BOOL hasProperDate = NO;
+
+  if (hasProperPrefix && hasProperSuffix) {
+    NSUInteger lengthOfMiddle = fileName.length - IDMLogFilePrefix.length - @".log".length;
+
+    // Date string should have at least 19 characters: "2013-12-03-17-14-10"
+    if (lengthOfMiddle >= 19) {
+      NSRange range = NSMakeRange(IDMLogFilePrefix.length, lengthOfMiddle);
+
+      NSString *middle = [fileName substringWithRange:range];
+      NSArray *components = [middle componentsSeparatedByString:@"-"];
+
+      if (components.count == 6) {
+        NSDateFormatter *dateFormatter = [self logFileDateFormatter];
+        NSDate *date = [dateFormatter dateFromString:middle];
+
+        if (date) {
+          hasProperDate = YES;
+        }
+      }
+    }
+  }
+
+  return (hasProperPrefix && hasProperDate && hasProperSuffix);
+}
+
+
+@end
+
+static NSString *const IDMLogFormatterDateFormat = @"yyyy-MM-dd HH:mm:ss.SSS";
+static NSString *const IDMLogFormatterDateFormatterKey = @"sh.calaba-IDMLogFormatter-NSDateFormatter";
+
+@interface IDMLogFormatter : NSObject <DDLogFormatter>  {
+  int32_t atomicLoggerCount;
+  NSDateFormatter *threadUnsafeDateFormatter;
```

Comment:
https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/Documentation/CustomFormatters.md#thread-safety-advanced

> threadunsafe

If there is exactly one logger, the thread unsafe date formatter will be used.  It is an optimization copied from the CocoaLumberjack docs.

If we wanted, for example, to have a TTY logger for the Xcode console, we would have 2 loggers which may or may not share a log formatter instance.   In that case, the thread safe date formatter would be used.

I started using this pattern when I was developing for iOS and had an HTTP logger, TTY, and file logger.  This class was copied directly from LPServer.  Maybe it is overkill, but it will protect maintainers from having very strange timestamps if we ever do have additional loggers.